.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+name: test-and-release
+
+on:
+  push:
+    branches: [ main ]
+  release:
+    types:
+      - created
+
+jobs:
+  release:
+    name: Release code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Log in to Quay
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Extract tags for Docker images
+        id: docker-meta
+        uses: docker/metadata-action@v3
+        with:
+          images: quay.io/aptible/www-prod
+
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.docker-meta.outputs.tags }}
+          build-args: |
+            CONTENTFUL_ACCESS_TOKEN=${{ secrets.CONTENTFUL_ACCESS_TOKEN }}
+            CONTENTFUL_SPACE_ID=${{ secrets.CONTENTFUL_SPACE_ID }}
+
+      # The `if` in this step and the following step are a bit ugly, but this is the cleanest
+      # way I found short of having two actions.
+      - name: Deploy code to staging
+        uses: aptible/aptible-deploy-action@v1
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        with:
+          username: ${{secrets.APTIBLE_ROBOT_USERNAME}}
+          password: ${{secrets.APTIBLE_ROBOT_PASSWORD}}
+          app: 'www-staging'
+          environment: 'aptible-staging'
+          docker_img: 'quay.io/aptible/www-prod:main'
+
+      - name: Deploy code to production
+        uses: aptible/aptible-deploy-action@v1
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
This looks right. I don't know if it's the standard best practice, but a few other examples suggest yes.